_episodes/06-raster-intro.md

```diff
@@ -0,0 +1,462 @@
+
+
+
+## Calculate Raster Statistics
+
+It is useful to know the minimum or maximum values of a raster dataset. In this
+case, given we are working with elevation data, these values represent the
+min/max elevation range at our site.
+
+We can compute these and other descriptive statistics with `min`, `max`, `mean`, and `std`.
+
+~~~
+print(raster_ams_b9.min())
+print(raster_ams_b9.max())
+print(raster_ams_b9.mean())
+print(raster_ams_b9.std())
+~~~
+{: .language-python}
+~~~
+<xarray.DataArray ()>
+array(0, dtype=uint16)
+Coordinates:
+    spatial_ref  int64 0
+<xarray.DataArray ()>
+array(15497, dtype=uint16)
+Coordinates:
+    spatial_ref  int64 0
+<xarray.DataArray ()>
+array(1652.44009944)
+Coordinates:
+    spatial_ref  int64 0
+<xarray.DataArray ()>
+array(2049.16447495)
+Coordinates:
+    spatial_ref  int64 0
+~~~
+{: .output}
+
+
+The information above includes a report of the min, max, mean, and standard deviation values, along with the data type.
+
+If we want to see specific quantiles, we can use xarray's `.quantile()` method. For example for the 25% and 75% quartiles:
+
+```python
+print(raster_ams_b9.quantile([0.25, 0.75]))
+```
+```
+<xarray.DataArray (quantile: 2)>
+array([   0., 2911.])
+Coordinates:
+  * quantile  (quantile) float64 0.25 0.75
+```
+{: .output}
+
+> ## Data Tip - NumPy methods
+> You could also get each of these values one by one using `numpy`.
+> 
+> ~~~
+> import numpy
+> print(numpy.percentile(raster_ams_b9, 25))
+> print(numpy.percentile(raster_ams_b9, 75))
+> ~~~
+> {: .language-python}
+> ~~~
+> 0.0
+> 2911.0
+> ~~~
+> {: .output}
+> You may notice that `raster_ams_b9.quantile` and `numpy.percentile` didn't require an argument specifying the axis or dimension along which to compute the quantile. This is because `axis=None` is the default for most numpy functions, and therefore `dim=None` is the default for most xarray methods. It's always good to check out the docs on a function to see what the default arguments are, particularly when working with multi-dimensional image data. To do so, we can use`help(raster_ams_b9.quantile)` or `?raster_ams_b9.percentile` if you are using jupyter notebook or jupyter lab.
+{: .callout}
+
+## Dealing with Missing Data
+So far: we have visualized a band of a Sentinel-2 scene, and calculated its statistics. However, there is one thing we also need to take into account: the missing value. As we have seen in the `raster_ams_b9.rio.nodata` field: the missing value is 0. Therefore when we plot the band data, or calculate the statistics, the missing value was not distinguished from the other values. Therefore, they may cause some unexpected results. For example, in the percentile we just calculated, the 25% percentile is still 0. This may simply refecting we have a lot of missing data in the raster.
+
+An option to distinguish missing values from real data, is to use `nan` values to represent them. This can be done by specifying `masked=True` when loading the raster:
```

Comment:
```suggestion
So far, we have visualized a band of a Sentinel-2 scene and calculated its statistics. However, we need to take missing values into account. We have seen that missing values are 0 in the `raster_ams_b9.rio.nodata` field. If we plot the data or calculate the statistics, missing values are not distinguished from the other values. This causes some unexpected results, for example, the 25% percentile that we just calculated, is 0. This shows that there are a lot of missing values in the data. To distinguish missing values from real data, we can use `nan` to represent them. This can be done by specifying `masked=True` when loading the raster:
```

Reply:
Thanks for both of your comments on this. I combined both of your suggestions and made an updated version. Also took Francesco's advice to use the word "missing data rather than" missing value".